src/utils/deep-merge.js

```diff
@@ -25,5 +35,5 @@ export default function deepMerge(
 }
 
 function isCloneable(obj: mixed) {
-  return Array.isArray(obj) || {}.toString.call(obj) == '[object Object]';
+  return typeof obj === 'object' || typeof obj === 'function';
```

Comment:
why `typeof obj === 'function'`? seems like you wouldn't want to call deepMerge with a function as one of the arguments.

Reply:
may be I didn't get it, but actually function is object and I want it to be called in deepMerge as a normal object

Reply:
if you check where `isClonable` is used, it's doing this:

```
if (isCloneable(value)) {
  target[key] = _deepMerge.apply(this, [target[key] || {}, value])
}
```

Imagine that `value` is a function like `() => true`, this means you're trying to make a call like:

```
deepMerge(obj, () => true)
```

this makes no sense. you should only be passing object/null/undefined to deepMerge.

What was wrong with the initial implementation of this method? It's what was used in `just-extend` which is battle tested in countless production apps. We should just keep that.